web/modules/custom/joinup_front_page/joinup_front_page.module

```diff
@@ -12,8 +12,21 @@ declare(strict_types = 1);
  */
 function joinup_front_page_preprocess_block(&$variables): void {
   if (\Drupal::currentUser()->isAuthenticated()) {
-    // The libraries are available for authenticated users.
-    return;
+    // The libraries are available for all users except the
+    // ones only with authenticated role.
+    $roles = [
+      'licence_manager',
+      'moderator',
+      'administrator',
+      'rdf_graph_manager',
+    ];
+
+    $user_roles = \Drupal::currentUser()->getRoles();
+    foreach ($roles as $role) {
+      if (in_array($role, $user_roles)) {
+        return;
+      }
+    }
```

Comment:
Is it possible to skip this whole role filtering? Is this perhaps needed to work around some bug or conflict?

It seems to me that all users regardless of role should be able to interact with the modal, the Discover Topics modal is intended for every user on Joinup.

Reply:
After several tests with different user permissions we do not need this rule.